src/main/java/com/google/starfish/services/FavoriteNoteService.java

```diff
@@ -115,6 +128,141 @@ public long getNumFavoritesByNoteId(DataSource pool, long noteId) throws SQLExce
+        }
+        rs.close();
+        return notes.toArray(new Object[0][0]);
+      }
+    }
+  }
+
+  /** Gets a sql where clause that filters by school or course if school or course is not null */
+  private String getStatementToFilterBySchoolOrCourse(String school, String course) {
+    String filterStmt = "WHERE 1=1 ";
+    if (school != null && !school.isEmpty()) {
+      filterStmt += "AND `school`= '" + school + "' ";
+    }
+    if (course != null && !course.isEmpty()) {
+      filterStmt += "AND `course`= '" + course + "' ";
+    }
+    return filterStmt;
+  }
+
+  /** Gets trending notes based on number of favorites in a given timespan */
+  private Object[][] getTrendingNotes(DataSource pool, Recency recency) throws SQLException {
+    Date date = getDateBasedOnRecency(recency);
+    List<Object> notes = new ArrayList<>();
+    try (Connection conn = pool.getConnection()) {
+      String stmt = getTrendingNotesStatement(null);
+      try (PreparedStatement selectStmt = conn.prepareStatement(stmt)) {
+        selectStmt.setDate(1, date);
+        ResultSet rs = selectStmt.executeQuery();
+        while (rs.next()) {
+          Note thisNote = constructNoteFromSqlResult(pool, rs);
+          Long numFavoritesInTimespan = rs.getLong(NUM_FAVORITES_IN_TIMESPAN_ID);
+          notes.add(new Object[] {thisNote, numFavoritesInTimespan});
+        }
+        rs.close();
+        return notes.toArray(new Object[0][0]);
+      }
+    }
+  }
+
+  /** Gets the sql query to return trending notes possibly filtered by school and/or course */
+  private String getTrendingNotesStatement(String schoolAndCourseFilter) {
+    String stmt = 
+        "SELECT * "
+      + "FROM " + NOTES + " AS a "
+        + "LEFT JOIN (SELECT note_id, COUNT(*) AS " + NUM_FAVORITES_IN_TIMESPAN_ID + " "
+                    + "FROM " + FAVORITE_NOTES + " "
+                    + "WHERE date >= ? " 
+                    + "GROUP BY note_id) AS b " 
+      + "ON a.id=b.note_id ";
+    if (schoolAndCourseFilter != null) stmt += schoolAndCourseFilter;
+    stmt += "ORDER BY count DESC;";
+    return stmt;
+  }
+
+  /** Gets the date based on recency */
+  private Date getDateBasedOnRecency(Recency recency) {
+    Date date;
+    Calendar calendar = Calendar.getInstance();
+    switch(recency) {
+      case TODAY:
+        date = new Date(calendar.getTimeInMillis());
+        break;
+      case THIS_WEEK:
+        calendar.add(Calendar.DAY_OF_MONTH, -7);
+        date = new Date(calendar.getTimeInMillis());
+        break;
+      case THIS_MONTH:
+        calendar.add(Calendar.DAY_OF_MONTH, -30);
+        date = new Date(calendar.getTimeInMillis());
+        break;
+      case ALL_TIME:
+        // No note should ever be favorited more than 100 years ago (for now...)
+        calendar.add(Calendar.YEAR, -100);
+        date = new Date(calendar.getTimeInMillis());
+        break;
+      default:
+        date = new Date(calendar.getTimeInMillis());
+    }
+    return date;
```

Comment:
Since you set `date` in every single case-- can you not just do that at the very end after leaving the switch block (after `calendar` has been added to when necessary)?

Reply:
Ohhh yep you're right, no need to set the value for date in the switch statement!